functions/src/submitApplication/constraints.js

```diff
@@ -115,4 +115,168 @@ exports.constraints = {
     },
     recaptchaResponse: atLeastOneCharacter,
   },
+  hacker_full_info: {
```

Comment:
If we're going to be validating the constraints on a per-collection basis instead of the form data as a whole, can we remove the generic `hacker` constraints?
https://github.com/nwplus/nwhacks2019/blob/47436eaa73bc5c5f8ee9007cf0a6d64b8209dbbb/functions/src/submitApplication/constraints.js#L25-L117

Reply:
No, we need all these constraints because we need to be able to distinguish between whether the input to the cloud function is valid (if invalid, error code is 400, meaning the user didn't pass in the correct data) vs whether the transformed data to store in firestore is valid (if invalid, error code is 500, meaning the server's fault).

Reply:
Oh got it, that makes sense